apps/server/src/controllers/ontimeController.ts

```diff
@@ -204,6 +204,7 @@ export const postSettings = async (req, res) => {
     const settings = DataProvider.getSettings();
     const editorKey = extractPin(req.body?.editorKey, settings.editorKey);
     const operatorKey = extractPin(req.body?.operatorKey, settings.operatorKey);
+    const serverPort = extractPin(req.body?.serverPort, settings.serverPort.toString());
```

Comment:
nice, few notes from our previous talks
- we should refuse the change if we are running in docker
- we should add this to the validation function

Do you think w could check if the port is available before making the change? If the user were to change this it could potentially "brick" the app.

Unsure if that should happen here or when the server attempts start up

Reply:
> Do you think w could check if the port is available before making the change? If the user were to change this it could potentially "brick" the app.

I don't know, do you expect the server to immediately reboot and listen on the new port when this is changed or is it ok to use this port when it is restarted manually by the user?

Reply:
I think as first step we would want the user to restart